From 55f1816a3b2fd31c9a2ca8ee7c01bcc387f1700e Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Tue, 19 Jul 2022 11:36:11 +0800
Subject: [PATCH] mm/sharepool: Fix kabi borken in sp_group_master
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 55f1816a3b2fd31c9a2ca8ee7c01bcc387f1700e
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DS9S
CVE: NA

-------------------------------------------------

The sp_group_master structure is used only in sharepool subsys and no
other drivers use it.

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/share_pool.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/share_pool.h b/include/linux/share_pool.h
index 654dc8cc2922..b1b81947fa3f 100644
--- a/include/linux/share_pool.h
+++ b/include/linux/share_pool.h
@@ -10,6 +10,7 @@
 #include <linux/hashtable.h>
 #include <linux/numa.h>
 #include <linux/jump_label.h>
+#include <linux/kabi.h>
 
 #define SP_HUGEPAGE		(1 << 0)
 #define SP_HUGEPAGE_ONLY	(1 << 1)
@@ -172,7 +173,7 @@ struct sp_group_master {
 	 * Used to apply for the shared pool memory of the current process.
 	 * For example, sp_alloc non-share memory or k2task.
 	 */
-	struct sp_group *local;
+	KABI_EXTEND(struct sp_group *local)
 };
 
 /*
-- 
2.33.0

